File: LPAv/Trabalho 9 - Python/questao10.py
import re

stringDeEntrada = input()
while (stringDeEntrada != "####"):
    if (re.match("^[A-Z]+[a-z]+[0-9]+([!@#$%^&*()_+={}\\\\|\\/?<>-]|[\\[\\]])+$", stringDeEntrada)):
        print("SIM")
    else:
        print("NAO")
    stringDeEntrada = input()

File: LPAv/Trabalho 9 - C++/questao10.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <regex>
using namespace std;

int main()
{
    string expressaoRegular;
    cin >> expressaoRegular;
    while (expressaoRegular != "####")
    {
        if (regex_match(expressaoRegular, regex("^[A-Z]+[a-z]+[0-9]+([!@#$%^&*()_+={}\\\\|\\/?<>-]|[\\[\\]])+$")))
        {
            printf("SIM\n");
        }
        else
        {
            printf("NAO\n");
        }
        cin >> expressaoRegular;
    }
}